galley/pkg/crd/validation/validation.go

```diff
@@ -77,14 +115,40 @@ func RunValidation(vc *WebhookParameters, printf, faltaf shared.FormatFn, kubeCo
 		validationLivenessProbe.RegisterProbe(livenessProbeController, "validationLiveness")
 		defer validationLivenessProbe.SetAvailable(errors.New("stopped"))
 	}
+
+	// Create the stop channel for all of the servers.
+	stop := make(chan struct{})
+
 	if readinessProbeController != nil {
 		validationReadinessProbe := probe.NewProbe()
-		validationReadinessProbe.SetAvailable(nil)
+		validationReadinessProbe.SetAvailable(errors.New("init"))
 		validationReadinessProbe.RegisterProbe(readinessProbeController, "validationReadiness")
-		defer validationReadinessProbe.SetAvailable(errors.New("stopped"))
+
+		go func() {
+			ready := false
+			for {
+				if err := webhookHTTPSHandlerReady(vc); err != nil {
+					validationReadinessProbe.SetAvailable(errors.New("not ready"))
+					scope.Infof("https handler for validation webhook  is not ready: %v", err)
```

Comment:
nit: extra space after webhook

Reply:
also, while working on codecov these days, I notice code like this can cause code coverage to flake, because this path may or may not happen depending on it becomes ready the first time.

not saying this is bad either, but ideally we may need a (unit) test to exercise the retry path for sure. =)

Reply:
Agreed. Verify little of the validation server bootstrap code has unit test coverage right now.